File: cpp/48c_1840A_CipherShifer.cpp
#include<bits/stdc++.h>

std::string solve(std::string s, int n){
    std::string sol = "";
    sol += s[0];
    char prev = s[0];
    for(int i = 1; i < (n-1); i++){
        if (s[i] == prev){
            sol += s[i+1];
            prev = s[i+1];
            i++;
        }
        //std::cout<<sol<<" "<<prev<<"\n";
    }
    return sol;
}

int main(){
    int t, n;
    std::string s;
    std::vector<std::string> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::cin>>s;
        ans.push_back(solve(s, n));
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/48c_1840A_CipherShifer.py
def solve(s, n):
    sol = ""
    sol += s[0]
    prev = s[0]
    flag = 0
    for i in range(1,n-1):
        if flag == 1:
            prev = s[i]
            sol += s[i]
            flag = 0
        elif s[i] == prev:
            flag = 1
    return sol

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        ans.append(solve(s, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()